internal/providers/amazon/objectstore.go

```diff
@@ -242,7 +242,7 @@ func (s *objectStore) deleteFailed(bucket *ObjectStoreBucketModel, reason error)
 	if err := s.db.Save(bucket).Error; err != nil {
 		return emperror.WrapWith(err, "failed to delete bucket", "bucket", bucket.Name)
 	}
-	return nil
+	return emperror.WrapWith(reason, "bucket", bucket.Name)
```

Comment:
this seems to be wrong here, if the bucket can't be deleted on the provider and the operation is not forced this code should not be reached, and the error returned by the provider is returned

Reply:
fixed - the original error is funneled through this method ...